pinot-core/src/test/java/org/apache/pinot/queries/MultiValueRawQueriesTest.java

```diff
@@ -338,6 +338,445 @@ public void testSelectQueries() {
+        Object[] values = recordRows.get(i);
+        assertEquals(values.length, 4);
+        assertEquals((int) values[0], expectedSVInts[i]);
+      }
+    }
+    {
+      // Test a group by order by query on some raw MV rows (order by on int and double)
+      String query = "SELECT mvRawIntCol, mvRawDoubleCol, mvRawStringCol from testTable GROUP BY mvRawIntCol, "
+          + "mvRawDoubleCol, mvRawStringCol ORDER BY mvRawIntCol, mvRawDoubleCol LIMIT 20";
+      ResultTable resultTable = getBrokerResponse(query).getResultTable();
+      assertNotNull(resultTable);
+      DataSchema dataSchema = new DataSchema(new String[]{
+          "mvRawIntCol", "mvRawDoubleCol", "mvRawStringCol"
+      }, new DataSchema.ColumnDataType[]{
+          DataSchema.ColumnDataType.INT, DataSchema.ColumnDataType.DOUBLE, DataSchema.ColumnDataType.STRING
+      });
+      assertEquals(resultTable.getDataSchema(), dataSchema);
+      List<Object[]> recordRows = resultTable.getRows();
+      assertEquals(recordRows.size(), 20);
+
+      int intValue = -1;
+      double doubleValue = -1;
+      for (int i = 0; i < 20; i++) {
+        Object[] values = recordRows.get(i);
+        assertEquals(values.length, 3);
+        int intValueCur = (int) values[0];
+        double doubleValueCur = (double) values[1];
+        assertTrue(intValueCur >= intValue);
+        if (intValueCur == intValue) {
+          assertTrue(doubleValueCur >= doubleValue);
+        }
+
+        intValue = intValueCur;
+        doubleValue = doubleValueCur;
+      }
+    }
+    {
+      // Test a group by order by query on some raw MV rows (order by on string)
+      String query = "SELECT mvRawIntCol, mvRawDoubleCol, mvRawStringCol from testTable GROUP BY mvRawIntCol, "
+          + "mvRawDoubleCol, mvRawStringCol ORDER BY mvRawStringCol LIMIT 10";
+      ResultTable resultTable = getBrokerResponse(query).getResultTable();
+      assertNotNull(resultTable);
+      DataSchema dataSchema = new DataSchema(new String[]{
+          "mvRawIntCol", "mvRawDoubleCol", "mvRawStringCol"
+      }, new DataSchema.ColumnDataType[]{
+          DataSchema.ColumnDataType.INT, DataSchema.ColumnDataType.DOUBLE, DataSchema.ColumnDataType.STRING
+      });
+      assertEquals(resultTable.getDataSchema(), dataSchema);
+      List<Object[]> recordRows = resultTable.getRows();
+      assertEquals(recordRows.size(), 10);
+
+      String stringVal = null;
+      for (int i = 0; i < 10; i++) {
+        Object[] values = recordRows.get(i);
+        assertEquals(values.length, 3);
+        String stringValueCur = (String) values[2];
+        if (stringVal != null && !stringValueCur.equals(stringVal)) {
+          assertTrue(stringValueCur.compareTo(stringVal) > 0);
+        }
+        stringVal = stringValueCur;
+      }
+    }
+    {
+      // Test a select with a VALUEIN transform function with group by
+      String query = "SELECT VALUEIN(mvRawIntCol, '0') from testTable WHERE mvRawIntCol IN (0) GROUP BY "
```

Comment:
Does this exercise the scenario I had talked about offline ? Some rows in your generated data should have MV column row / array with value 0 or something else for VALUEIN to remove them during transform ?

Reply:
yes only a subset of rows have value '0' in the column. in such rows, only one value is '0' and the other value is different.